azure/services/async/async.go

```diff
@@ -68,6 +68,36 @@ func processOngoingOperation(ctx context.Context, scope FutureScope, client Futu
 	return result, err
 }
 
+// GetExistingResource implements the logic for creating a resource.
+func GetExistingResource(ctx context.Context, scope FutureScope, client Creator, spec azure.ResourceSpecGetter, serviceName string) (interface{}, error) {
```

Comment:
does this need to be exported? If it's just a helper for CreateResource, it should start with a lower case